node/linux/datadog-aas-node-linux/datadog_wrapper

```diff
@@ -1,44 +1,76 @@
-#!/usr/bin/env sh 
+#!/usr/bin/env sh
+
+CURRENT_DIR=$(pwd)
+
+if [ -z "$DD_DIR" ]; then
+    DD_DIR="/home/datadog"
+fi
+
+if [ -z "$DD_BINARIES_URL" ]; then
+    DD_BINARIES_URL="https://gist.github.com/jcstorms1/7e73ab0da121f42d1843dac5b75c21cb/raw"
+fi
+
+setUpCommonEnv() {
+    echo "Creating datadog directory"
+    mkdir -p $DD_DIR && cd $DD_DIR || return
+
+    echo "Downloading required datadog binaries from $DD_BINARIES_URL"
+    curl -L $DD_BINARIES_URL/datadog.tar.gz -o datadog.tar.gz || return
+
+    echo "Decompressing files"
+    tar -xzf datadog.tar.gz || return
+    rm datadog.tar.gz
+
+    echo "Starting trace agent"
+    ./trace-agent &
+    TRACER_PID=$!
+    trap 'kill $TRACER_PID ; exit' INT
+}
 
 setUpNodeEnv() {
     echo "Setting up Datadog tracing for Node"
-    DD_DIR="/home/site/wwwroot/datadog-js"
-    mkdir -p $DD_DIR
 
-    curl -L "https://gist.github.com/jcstorms1/7e73ab0da121f42d1843dac5b75c21cb/raw/datadog-js.tar.gz" -o $DD_DIR/datadog-js.tar.gz
-    tar -xzf $DD_DIR/datadog-js.tar.gz && rm $DD_DIR/datadog-js.tar.gz
-    
-    export NODE_OPTIONS="--require=$DD_DIR/node_modules/dd-trace/init --require=$DD_DIR/start-trace-agent.js"
+    setUpCommonEnv || return
+
+    echo "Installing Node tracer"
+    npm install --prefix $DD_DIR dd-trace || return
+
+    ORIG_NODE_OPTIONS=$NODE_OPTIONS
+    export NODE_OPTIONS="--require=$DD_DIR/node_modules/dd-trace/init $ORIG_NODE_OPTIONS"
+
+    # confirm updates to NODE_OPTIONS
+    node --help >/dev/null || (export NODE_OPTIONS=$ORIG_NODE_OPTIONS && return)
```

Comment:
It is possible that our update to the `NODE_OPTIONS` could crash their app. This happens for example if the `dd-trace/init` file does not exist. This line here checks to confirm that node can in fact start up using the given `NODE_OPTIONS`. If it can't, we reset the var and return.